### ISSUE TITLE
Translations with M2M relation do not get saved

### ISSUE THREAD

**Issue by blackfox00** [directus/api#1007]
# Bug Report

When having a collection that has translations relation and that translation relation is having M2M relation, any fields in the translation relation do not get saved on update if there is any M2M relation item selected. If the M2M relation interface is empty (no item selected), it gets saved OK

## Steps to Reproduce

1. Create 4 collections with the following or similar schema: 
https://dbdiagram.io/d/5cff47acfff7633dfc8e2536

2. Create a new blog item, fill in the fields (title, translated title, translated content), create a new tag and save. Go back to edit item end edit translated content, press save

## Expected Behavior

Updated fields in translated content get saved

## Actual Behavior
Updated fields in translated content do not get saved, no error is thrown (success response with original data)

## Other Context & Screenshots

Here are the links for request and response on the PATCH call, highlighted is the updated field in request and then in response there is still the original data (note that there are more fields in this setup than provided schema but essentially it's the same) 
 
https://i.imgur.com/99AwAe8.png
https://i.imgur.com/5r6OQhH.png

## Technical Details 
- API 2.1.1, APP 7.5.0
- Device: Desktop
- OS: Alpine 3.8
- Web Server: nginx 1.14.2
- PHP Version: 7.2.18
- Database: 5.7.26
- Install Method: docker (updated to pull newest versions)

**Comment by blackfox00** [comment]
It also affects One to Many relations with the same behaviour. Another obesrvation is that the `delta` of made changes are saved in `directus_revisions`  but not the changes itself in respective table

**Comment by itsmerhp** [comment]
Hello @benhaynes , @rijkvanzanten 

This issue is taking place due to translation details is not updated in nested request payload(in M2M relation field) of blog record update.
In this case, **blog_translation** collection's M2M relation is defined with **tag** collection, so in update API the translation payload will also passed in M2M field if **tag** record is already selected along with translation details. 

For example : 
> Blog record has been created with translation details and tag record has been created/selected existing in it. **tag** collection is associated with **blog_translation** collection by M2M relationship.
> On update blog record, change translation details but keep associate tag record as it is, so on update API call request payload is as below:

```
{
	"translations": [{
		"id": 3,
		"language_code": "en",
		"title": "Test - Update",
		"content": "Test - Update",
		"blog_tag": [{
			"id": 7,
			"blog_translation_id": {
				"id": 3,
				"blog_id": "3",
				"language_code": "en",
				"title": "Test",
				"content": "Test"
			},
			"tag_id": {
				"id": 1,
				"name": "BC-1"
			}
		}]
	}]
}
```
so, in above payload translation details update properly in translations field, but in M2M field payload translation details mapped with **blog_translation_id** field is not updated in request param. so, due to old translation details in M2M relation field payload, it is resetting translation data. Therefore, to avoid resetting translation data in update, it needs to be update in request nested payload in API call from APP side.

In summary, translation details passed twice in request payload, one in main payload and second in M2M field nested payload, main payload translation details passing is updated and in M2M nested payload old translation details is passing.

Please share your thoughts.

**Comment by bjgajjar** [comment]
Hey @rijkvanzanten 

Let's move this issue to `APP` as @itsmerhp suggested. :)
